### PR TITLE
fix(ui): add groups / users bugs and badge ui bug

### LINKED ISSUE #2831: ui: creating a user with a duplicate email doesn't show error
Screen Shot 2022-08-04 at 3 38 04 PM

### LINKED ISSUE #2833: ui: `x` icon on badge is difficult to click
On the email badges, the `x` button is difficult to click due to its small size. We should add some padding to give it a larger click radius

Screen Shot 2022-08-04 at 4 23 35 PM

### LINKED ISSUE #2839: ui: ability to create empty groups

### PR DESCRIPTION
**Pull request by hoyyeva**
## Related Issues
Resolves #2831 
Resolves #2833 
Resolves #2839

### CODE REVIEW

File: ui/pages/users/add.js [hunk 1]
```diff
@@ -123,6 +123,16 @@ export default function UsersAdd() {
           errors[error.fieldName.toLowerCase()] =
             error.errors[0] || 'invalid value'
         }
+
+        // TODO: need to work with backend for better error message
+        if (
+          e.code === 409 &&
+          errors['identity_id'] &&
```
Comment (jmorganca) [inline ui/pages/users/add.js:130]: `errors.identity_id`

File: ui/pages/users/add.js [hunk 2]
```diff
@@ -123,6 +123,16 @@ export default function UsersAdd() {
...
+          errors['identity_id'].includes('already exist')
```
Comment (jmorganca) [inline ui/pages/users/add.js:131]: Error 409 implies a conflict so I think we don't have to check for this string (which could be flaky)

File: ui/pages/users/add.js [hunk 3]
```diff
@@ -123,6 +123,16 @@ export default function UsersAdd() {
...
+        ) {
+          errors['name'] = 'user already exists'
```
Comment (jmorganca) [inline ui/pages/users/add.js:133]: `errors.name =`

**Review by jmorganca — COMMENTED** [review]
Looking good! A few comments ahead of merging

File: ui/components/badge.js [hunk 4]
```diff
@@ -2,16 +2,18 @@ import { XIcon } from '@heroicons/react/outline'
 
 export default function Badge({ children, onRemove }) {
   return (
-    <div className='my-1 mr-1 flex items-center justify-center overflow-hidden text-ellipsis rounded-md bg-gray-800 py-1 px-2 font-medium text-white'>
-      <div className='max-w-full flex-initial overflow-hidden text-ellipsis text-xs font-normal leading-none'>
+    <div className='my-1 mr-1 flex items-center justify-center overflow-hidden text-ellipsis rounded-md bg-gray-800 font-medium text-white first:my-0'>
```
Comment (hoyyeva) [inline ui/components/badge.js:5]: to prevent input changes size between being empty and having 1+ badges

Comment (jmorganca) [reply ui/components/badge.js:5]: nice!

File: ui/pages/users/add.js [hunk 5]
```diff
@@ -123,6 +123,12 @@ export default function UsersAdd() {
           errors[error.fieldName.toLowerCase()] =
             error.errors[0] || 'invalid value'
         }
+
+        // TODO: need to work with backend for better error message
+        if (e.code === 409 && errors.identity_id) {
+          errors.name = 'user already exists'
```
Comment (jmorganca) [inline ui/pages/users/add.js:129]: 🤩

**Review by jmorganca — APPROVED** [review]
LGTM

**Pull request merged**